week2_shoppinglist/index.js

```diff
@@ -0,0 +1,384 @@
+  },
+  {
+    id: 8,
+    category: "digital",
+    image: "./images/usb.jpg",
+    name: "로이체 잔망루피 5in1 멀티 USB 허브",
+    price: 36900,
+    liked: false,
+  },
+  {
+    id: 9,
+    category: "digital",
+    image: "./images/mic.jpg",
+    name: "로이체 잔망루피 블루투스 마이크 스피커",
+    price: 39900,
+    liked: false,
+  },
+];
+
+// 상품 리스트 렌더링
+function renderItems(filterCategory = "all") {
+  const filteredItems = SHOPPING_LIST.filter((item) => {
+    return filterCategory === "all" || item.category === filterCategory; // 상품 카테고리 필터링
+  });
+
+  const itemsContainer = document.querySelector(".items"); // 상품 리스트를 렌더링할 class 가져오기
+  itemsContainer.innerHTML = ""; // 기존 상품 목록 비우기
+
+  // 상품 목록 추가
+  filteredItems.forEach((item) => {
+    const article = document.createElement("article"); // 각 상품에 대한 article 생성
+    article.className = "item";
+    article.onclick = function () {
+      addCart(item);
+    };
+
+    // 이미지
+    const img = document.createElement("img");
+    img.src = item.image;
+    img.alt = "상품 이미지";
+    img.className = "item-image";
+
+    // 좋아요
+    const likeDiv = document.createElement("div");
+    likeDiv.className = "like";
+
+    // 상품명
+    const nameP = document.createElement("p");
+    nameP.className = "name";
+    nameP.textContent = item.name;
+
+    // 가격
+    const priceP = document.createElement("p");
+    priceP.className = "price";
+    priceP.textContent =
+      item.price.toString().replace(/\B(?<!\.\d*)(?=(\d{3})+(?!\d))/g, ",") +
+      "원"; // 천단위 표시
+
+    // 요소 추가
+    article.appendChild(img);
+    article.appendChild(likeDiv);
+    article.appendChild(nameP);
+    article.appendChild(priceP);
+    itemsContainer.appendChild(article);
+  });
+}
+
+//장바구니 목록 추가
+function addCart(item) {
+  let cartList = JSON.parse(localStorage.getItem("cartList")) || []; // cartList가 배열이 아닌 경우, 빈 배열로 초기화
+  cartList.push(item); // 새 상품 추가
+  localStorage.setItem("cartList", JSON.stringify(cartList)); // 변경된 장바구니 목록을 로컬 스토리지에 저장
+  alert(`${item.name}을(를) 장바구니에 담았습니다.`);
+  var answer = confirm("장바구니로 이동하시겠습니까?");
+  if (answer == true) {
+    location.href = "cart.html";
+  }
+}
+
+/* cart.html */
+
+// 장바구니 목록 렌더링
+function renderCartItems() {
+  const cartItemsContainer = document.querySelector(".cartlist"); // 상품 리스트를 렌더링할 class 가져오기
+  cartItemsContainer.innerHTML = ""; // 기존 상품 목록 비우기
+
+  let cartList = JSON.parse(localStorage.getItem("cartList") || "[]");
+  cartList.forEach((item) => {
+    const tr = document.createElement("tr");
+    // 데이터셋에 아이템 속성을 개별적으로 추가
+    tr.dataset.id = item.id;
+    tr.dataset.name = item.name;
+    tr.dataset.image = item.image;
+    tr.dataset.price = item.price;
+    tr.dataset.category = item.category;
+
+    // 체크박스
+    const checktd = document.createElement("td");
+    const checkbox = document.createElement("input");
+    checkbox.type = "checkbox";
+    checkbox.name = "item";
+    checkbox.onclick = function () {
+      onclick = checkSelectAll(this);
+    };
```

Comment:
p5) this 키워드는 화살표 함수, 일반 함수에 따라서 바인딩이 달라질 수 있어서 위험할 수도 있는데요~ 여기서는 event.target을 사용하시는건 어떠세요~~? 

```suggestion
checkbox.onclick = function (event) {
      onclick = checkSelectAll(event.target);
    };
```